### PR TITLE
Fixed 500 reported for client disconnection

### PR DESCRIPTION
**Pull request by CoufalJa**
# What does this PR do? 

Fix an issue with random 500s reported in Traefik metrics middleware. Issue: https://github.com/containous/traefik/issues/1926

# Motivation
By default oxy middleware reports 500 for Context cancel error, which does not make much sense as in this case the 500 won't be received by the client and further HttpHandlers down the line (like Traefik metrics middleware) can go nuts because of it. I originally had this issue in Traefik so if it doesn't make sense to fix it in oxy I need to find a way how to fix it in Traefik only.

# More
Fixes: https://github.com/containous/traefik/issues/1926

### CODE REVIEW

**Comment by CoufalJa** [comment]
@Juliens Any feedback on this? I am not sure if this is a right direction but I am really keen on getting some fix for those 500s into traefik 1.7 otherwise we need to run from our fork. The issue might be bigger for us as most of our clients access our servers over unreliable networks where disconnections are not uncommon.

**Comment by Juliens** [comment]
I think, it's a good idea to catch the context.Canceled, but like this, in the access logs we got a 200, and I'm not sure it's a good idea to have 200 WDYT?

**Comment by Juliens** [comment]
I prefer a solution with a specific status code, but I think this is more a 4x than a 5x.

**Comment by CoufalJa** [comment]
Thanks, I have picked 499 the same as nginx does.
https://httpstatuses.com/499

File: utils/handler.go [hunk 1]
```diff
@@ -21,18 +28,27 @@ type StdHandler struct{}
...
 		} else {
 			statusCode = http.StatusBadGateway
+			statusText = http.StatusText(statusCode)
 		}
 	} else if err == io.EOF {
 		statusCode = http.StatusBadGateway
+		statusText = http.StatusText(statusCode)
+	} else if err == context.Canceled {
+		statusCode = StatusClientClosedRequest
+		statusText = StatusClientClosedRequestText
```
Comment (Juliens) [inline utils/handler.go]: WDYT about create 
```golang
func statusText(statusCode int) string {
    if statusCode == StatusClientClosedRequest {
        return StatusClientClosedRequestText
    }
    return http.StatusText(statusCode)
}
```
and use this new func in the `Write`?

**Pull request merged**